app/src/main/java/org/thoughtcrime/securesms/database/Storage.kt

```diff
@@ -1265,6 +1265,16 @@ open class Storage(
             }
             setRecipientHash(recipient, contact.hashCode().toString())
         }
+
+        // if we have contacts locally but that are missing from the config, remove their corresponding thread
+        val  removedContacts = getAllContacts().filter { localContact ->
+            moreContacts.firstOrNull {
+                it.id == localContact.accountID
+            } == null
+        }
+        removedContacts.forEach {
+            getThreadId(fromSerialized(it.accountID))?.let(::deleteConversation)
```

Comment:
I'm wondering if we should also remove other data. Like the one in `ProfileManager`. If you look at the code in the same function, back in around line 1233, new contacts get added into ProfileManager also.

Reply:
I think the process of a thread deletion takes that into account eventually in `threadDeleted`.
What do you think?